ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConvertJoinMapJoin.java

```diff
@@ -714,6 +737,23 @@ private boolean convertJoinBucketMapJoin(JoinOperator joinOp, OptimizeTezProcCon
       }
     }
 
+    if (bucketFunction != null) {
+      final Operator<?> bigTableOp = mapJoinOp.getParentOperators().get(bigTablePosition);
+      for (TableScanOperator tso : OperatorUtils.findOperatorsUpstream(bigTableOp, TableScanOperator.class)) {
+        tso.getConf().setGroupingPartitionColumns(bucketFunction.getSourceColumnNames());
+        tso.getConf().setGroupingNumBuckets(bucketFunction.getNumBuckets());
+      }
+
+      for (Operator<?> op : mapJoinOp.getParentOperators()) {
+        if (!(op instanceof ReduceSinkOperator)) {
```

Comment:
you could use stream API, filter and then foreach

Reply:
Done
https://github.com/apache/hive/pull/5409/commits/ae7444e8e1c2de95bfc1915af3d3c0245af4b472